ioengine/IOEngine.py

```diff
@@ -0,0 +1,53 @@
+"""The IOEngine class."""
+import inspect
+from types import ModuleType
+import sys
+from utils import dotdict
+
+
+class IOEngine:
+    """The IOEngine class"""
+
+    def __init__(self):
+        self.services = []
+        self.engine_api = dotdict(showMessage=print, register=self.register_service)
+
+        # make a fake module with api
+        mod = ModuleType("bioimage")
+        sys.modules[mod.__name__] = mod  # pylint: disable=no-member
+        mod.__file__ = mod.__name__ + ".py"  # pylint: disable=no-member
+        mod.api = self.engine_api
+
+    def register_service(self, **api):
+        """Set interface."""
+        if isinstance(api, dict):
+            api = {a: api[a] for a in api if not a.startswith("_")}
+        elif inspect.isclass(type(api)):
+            api = {a: getattr(api, a) for a in dir(api) if not a.startswith("_")}
+        else:
+            raise Exception("unsupported api export")
```

Comment:
Please implement a library specific exception with a library specific base class.
```suggestion
            raise UnsupportedAPI("Unsupported api export")
```

Put the exceptions in another module named `exceptions.py`:

```py
class IOEngineException(Exception):
    """Represent the base IOEngine exception."""


class UnsupportedAPI(IOEngineException):
    """Represent an unsupported api exception."""
```